### PR TITLE
Add TarLz4Util to improve perf for comressing snapshot

### PR DESCRIPTION
**Pull request by Linary**
Change-Id: I6d8d4d180a2216ad3c026e884e9e05bd6ed80eff

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import net.jpountz.lz4.LZ4BlockInputStream;
+import net.jpountz.lz4.LZ4BlockOutputStream;
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4Factory;
+import net.jpountz.lz4.LZ4FastDecompressor;
+
+/**
+ * Reference: https://mkyong.com/java/how-to-create-tar-gz-in-java/
+ */
+public final class TarLZ4Util {
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java:50]: can we move all the methods of ZipUtil and TarLZ4Util to CompressUtil

File: hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    public static void compress(String rootDir, String sourceDir,
+                                String outputFile, Checksum checksum)
+                                throws IOException {
+        LZ4Factory factory = LZ4Factory.fastestInstance();
+        LZ4Compressor compressor = factory.fastCompressor();
+        try (FileOutputStream fos = new FileOutputStream(outputFile);
+             CheckedOutputStream cos = new CheckedOutputStream(fos, checksum);
+             BufferedOutputStream bos = new BufferedOutputStream(cos);
+             LZ4BlockOutputStream lz4os = new LZ4BlockOutputStream(bos, 8192,
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java:60]: define const var for 8192

File: hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        return topLevel.getFileName().resolve(topLevel.relativize(current))
+                       .toString();
+    }
+
+    public static void decompress(String sourceFile, String outputDir,
+                                  Checksum checksum) throws IOException {
+        Path source = Paths.get(sourceFile);
+        Path target = Paths.get(outputDir);
+        if (Files.notExists(source)) {
+            throw new IOException("File doesn't exists");
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java:121]: add the source to error message

File: hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        }
+        LZ4Factory factory = LZ4Factory.fastestInstance();
+        LZ4FastDecompressor decompressor = factory.fastDecompressor();
+        try (InputStream fis = Files.newInputStream(source);
+             CheckedInputStream cis = new CheckedInputStream(fis, checksum);
+             BufferedInputStream bis = new BufferedInputStream(cis);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/util/TarLZ4Util.java:127]: what's the best buffer size

**Review by javeme — CHANGES_REQUESTED** [review]

File: hugegraph-core/src/main/java/com/baidu/hugegraph/util/CompressUtil.java [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4Factory;
+import net.jpountz.lz4.LZ4FastDecompressor;
+
+public final class CompressUtil {
+
+    /**
+     * Reference: https://mkyong.com/java/how-to-create-tar-gz-in-java/
+     */
+    public static void tarCompress(String rootDir, String sourceDir,
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/util/CompressUtil.java:62]: prefer compressTar

File: hugegraph-core/src/main/java/com/baidu/hugegraph/util/CompressUtil.java [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                                               .toString());
+                FileUtils.forceMkdir(entryFile.getParentFile());
+                try (FileOutputStream fos = new FileOutputStream(entryFile);
+                     BufferedOutputStream bos = new BufferedOutputStream(fos)) {
+                    IOUtils.copy(zis, bos);
+                    bos.flush();
+                    fos.getFD().sync();
+                }
+            }
+            // Continue to read all remaining bytes(extra metadata of ZipEntry)
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/util/CompressUtil.java:234]: prefer /* */

File: hugegraph-core/src/main/java/com/baidu/hugegraph/util/CompressUtil.java [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+             CheckedOutputStream cos = new CheckedOutputStream(fos, checksum);
+             BufferedOutputStream bos = new BufferedOutputStream(cos);
+             ZipOutputStream zos = new ZipOutputStream(bos)) {
+            CompressUtil.compressDirToZipFile(rootDir, sourceDir, zos);
+            zos.flush();
+            fos.getFD().sync();
+        }
+    }
+
+    private static void compressDirToZipFile(String rootDir,
```
Comment (zhoney) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/util/CompressUtil.java:194]: prefer zipDir

**Review by zhoney — CHANGES_REQUESTED** [review]

**Review by javeme — APPROVED** [review]

**Review by houzhizhen — APPROVED** [review]

**Pull request merged**